Code/Tools/FBuild/FBuildCore/Helpers/ToolManifest.cpp

```diff
@@ -117,6 +120,11 @@ bool ToolManifest::Generate( const Node * mainExecutable, const Dependencies & d
         *pos = xxHash::Calc32( relativePath );
         ++pos;
     }
+	for (size_t i = 0; i < numEnvVars; ++i)
+	{
+		*pos = xxHash::Calc32(m_CustomEnvironmentVariables[i]);
+		++pos;
+	}
```

Comment:
I don't think we need to include the custom environment variables in the hash, since they don't impact the layout of the directory at all.  The env vars should be correctly set for each remote execution, even if two different builders use the same toolchain.
Is there an edge case I'm missing here?